File: Restaurant/Restaurant.Web/Modules/ServerTypes/Default/ItemColumns.ts
import { ImageViewerFormatter } from "@/Default/Product/ImageViewerFormatter";
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ItemRow } from "./ItemRow";

export interface ItemColumns {
    Id: Column<ItemRow>;
    ItemName: Column<ItemRow>;
    Photo: Column<ItemRow>;
    Description: Column<ItemRow>;
    MealsName: Column<ItemRow>;
}

export class ItemColumns extends ColumnsBase<ItemRow> {
    static readonly columnsKey = 'Default.Item';
    static readonly Fields = fieldsProxy<ItemColumns>();
}

[ImageViewerFormatter]; // referenced types

File: Restaurant/Restaurant.Web/Modules/Default/Item/ItemColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Restaurant.Default.Columns;

[ColumnsScript("Default.Item")]
[BasedOnRow(typeof(ItemRow), CheckNames = true)]
public class ItemColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId")]
    public int Id { get; set; }
    [EditLink]
    public string ItemName { get; set; }
    [ImageViewerFormatter]
    public string Photo { get; set; }
    public string Description { get; set; }
    public string MealsName { get; set; }
}